applications/GeoMechanicsApplication/custom_processes/apply_k0_procedure_process.cpp

```diff
@@ -81,67 +165,65 @@ bool ApplyK0ProcedureProcess::UseStandardProcedure() const
     return !mSettings.Has(setting_name) || mSettings[setting_name].GetBool();
 }
 
-void ApplyK0ProcedureProcess::CalculateK0Stresses(Element& rElement)
+array_1d<double, 3> ApplyK0ProcedureProcess::CreateK0Vector(const Element::PropertiesType& rProp)
 {
-    // Get K0 material parameters of this element ( probably there is something more efficient )
-    const Element::PropertiesType& rProp             = rElement.GetProperties();
-    const int                      k0_main_direction = rProp[K0_MAIN_DIRECTION];
-    if (k0_main_direction < 0 || k0_main_direction > 1) {
-        KRATOS_ERROR << "undefined K0_MAIN_DIRECTION in ApplyK0ProcedureProcess: " << k0_main_direction
-                     << std::endl;
-    }
-
     // Check for alternative K0 specifications
     array_1d<double, 3> k0_vector;
     if (rProp.Has(K0_NC)) {
         std::fill(k0_vector.begin(), k0_vector.end(), rProp[K0_NC]);
     } else if (rProp.Has(INDEX_OF_UMAT_PHI_PARAMETER) && rProp.Has(NUMBER_OF_UMAT_PARAMETERS) &&
```

Comment:
Used the all 3 ( INDEX_OF_UMAT_PHI_PARAMETER, NUMBER_OF_UMAT_PARATERS, UMAT_PARAMETERS ) to show that they form a group together. Indeed it has been checked before.
Don't know what is best, as I even find NUMBER_OF_UMAT_PARAMETERS an undesired thing, because the length should be given by checking UMAT_PARAMETERS itself. That is something outside this PR.